pkg/R/types.R

```diff
@@ -0,0 +1,47 @@
+# Utility functions for handling Spark DataTypes.
+
+# Handler for StructType
+structType <- function(st) {
+  obj <- structure(new.env(parent = emptyenv()), class = "structType")
+  obj$jobj <- st
+  obj$fields <- lapply(SparkR:::callJMethod(st, "fields"), structField)
```

Comment:
This might be over-optimizing things, but it would be cool if these fields were populated lazily. i.e. only when the user wants to look at `fields` we actually make the `callJMethod`. One way to do that is to make these variables closures (i.e. `fields <- function() { lapply(SparkR:::callJMethod(st, "fields"), structField) }`) and users can call `st$fields()`  to get the fields.
@davies @concretevitamin Any thoughts ?